config/ompi_check_pvfs2.m4

```diff
@@ -46,70 +42,75 @@ AC_DEFUN([OMPI_CHECK_PVFS2],[
              [Build Pvfs2 support, optionally adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries])])
     OPAL_CHECK_WITHDIR([pvfs2], [$with_pvfs2], [include/pvfs2.h])
 
-    AC_ARG_WITH([pvfs2-libs],
-        [AC_HELP_STRING([--with-pvfs2-libs=LIBS],
-                       [Libraries to link with for pvfs2])])
-
-    temp_with_pvfs2="$with_pvfs2"
     AS_IF([test -z "$with_pvfs2"],
-          [with_pvfs2="/usr/local"])
-
-    temp_with_pvfs2_libs="$with_pvfs2_libs"
-    AS_IF([test -z "$with_pvfs2_libs"],
-	[with_pvfs2_libs="pvfs2 pthread"])
+          [ompi_check_pvfs2_dir="/usr/local"],
+          [ompi_check_pvfs2_dir=$with_pvfs2])
+
+# check for name of pthread library
+
+   AC_MSG_CHECKING([for library required to link pthread code])
+   cat > conftest.c <<EOF
+#include <pthread.h>
+void *tfunc (void *arg){
+ pthread_t tid;
+ tid = pthread_self();
+ return NULL;
+}
+
+int main ( int argc, char **argv )
+{
+ int ret, *val;
+ pthread_t threadid;
+ ret = pthread_create (&threadid, NULL, tfunc, NULL);
+ pthread_join( threadid, (void **)&val);
+ return 0;
+}
+EOF
+
+   # Try the compile
+   OPAL_LOG_COMMAND(
+      [$CC $CFLAGS $LDFLAGS conftest.c -o conftest -lpthread $LIBS],
+      [pthread_lib="-lpthread"],
+      []
+    )
+   rm -f conftest
+   OPAL_LOG_COMMAND(
+      [$CC $CFLAGS $LDFLAGS conftest.c -o conftest -lpthreads $LIBS],
+      [pthread_lib="-lpthreads"],
+      []
+    )
+   rm -f conftest
+   OPAL_LOG_COMMAND(
+      [$CC $CFLAGS $LDFLAGS conftest.c -o conftest -llthread $LIBS],
+      [pthread_lib="-llthread"],
+      []
+    )
+   rm -f conftest
+   OPAL_LOG_COMMAND(
+      [$CC $CFLAGS $LDFLAGS conftest.c -o conftest $LIBS],
+      [pthread_lib=""],
+      []
+    )
+    AC_MSG_RESULT([$pthread_lib])
+
+    rm -f conftest.c conftest
+    if test -e "$ompi_check_pvfs2_dir/lib64" ; then
+        ompi_check_pvfs2_libdir="$ompi_check_pvfs2_dir/lib64"
+    else
+        ompi_check_pvfs2_libdir="$ompi_check_pvfs2_dir/lib"
+    fi
 
     # Add correct -I and -L flags
-    AS_IF([test -d "$with_pvfs2/include"],
-        [check_pvfs2_CPPFLAGS="-I$with_pvfs2/include"
-            $1_CPPFLAGS="$check_pvfs2_CPPFLAGS"
-            CPPFLAGS="$CPPFLAGS $check_pvfs2_CPPFLAGS"],
-	[ompi_check_pvfs2_happy="no"])
+    OPAL_CHECK_PACKAGE([$1], [pvfs2.h], [pvfs2], [PVFS_util_resolve], [$pthread_lib],
```

Comment:
@edgargabriel I checked a little further: there's several variables:

* `$PTHREAD_CPPFLAGS`
* `$PTHREAD_CXXCPPFLAGS`
* `$PTHREAD_CFLAGS`
* `$PTHREAD_CXXFLAGS`
* `$PTHREAD_FCFLAGS`
* `$PTHREAD_LIBS`

The pthread setup is run before any of the MCA component configury, so these variables should definitely be available to you.  Some/all may be empty, but I'll bet that in your case, at least `$PTHREAD_CFLAGS` should contain something (e.g., `-pthread`?).